sdk/python/kfp/client/client.py

```diff
@@ -1650,3 +1651,26 @@ def validate_pipeline_resource_name(name: str) -> None:
         raise ValueError(
             f'Invalid pipeline name: "{name}". Pipeline name must conform to the regex: "{REGEX}".'
         )
+
+
+def _validate_pipeline_arguments(pipeline_obj: Dict[str, Any],
+                                 arguments: Mapping[str, Any]) -> None:
+    """Validates that all arguments without default values provided in the
+    pipeline_obj pipeline definition are included in arguments.
+
+    Args:
+        pipeline_obj (Dict[str, Any]): Dict object parsed from the yaml file.
+        arguments (Mapping[str, Any]): Arguments to the pipeline function.
+    """
+
+    input_definitions = pipeline_obj['root']['inputDefinitions']
+    parameters = input_definitions.get('parameters', {})
+    # note: artifact inputs are not supported at outermost pipeline level
```

Comment:
FYI, this is changing from https://github.com/kubeflow/pipelines/pull/8044

Reply:
Thanks, Chen. Let's merge #8044 first, then come back to this one. I will mark this as a draft in the meantime.